### DISCUSSION TITLE
TexStudio `.app` file has unfriendly name in ARM machines

### DISCUSSION THREAD

**Discussion by mateusfccp** [Homebrew/discussions#5439] [Casks]
### Output of `brew config`

```shell
mateusfccp@mateusfccp ~> brew config
HOMEBREW_VERSION: 4.3.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3f08c75e7b950d4340dab462f3e7f77e8093fa2b
Last commit: 2 days ago
Core tap JSON: 12 Jun 18:28 UTC
Core cask tap JSON: 12 Jun 18:28 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 11
Homebrew Ruby: 3.3.2 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.2/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.45.2 => /opt/homebrew/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
mateusfccp@mateusfccp ~> brew doctor
Your system is ready to brew.
```

### Description of issue

After installing TexStudio through the command `brew install texstudio', the `.app` file has the name `texstudio-4.8.0-osx-m1.app`.

I had a MacBook Pro 2019 Intel where this issue didn't happen and the `.app` file was named `TexStudio.app`.

Recently, I bought a M3 Pro, and after installing TexStudio, this unfriendly name was used instead.

<img width="1512" alt="image" src="https://github.com/Homebrew/discussions/assets/4605213/7149ea37-2871-41a2-bac9-19c1c873f9a0">

I tried to rename the `.app` file manually, and it worked flawlessly until I had to update it through `brew update`. In this case, `brew` searches for the original file name, and as it doesn't find, it throws an error.

### Relevant casks

texstudio (https://github.com/Homebrew/homebrew-cask/blob/619457413fc2155a8a4fd89a04338ee255a291d6/Casks/t/texstudio.rb)

**Comment by p-linnane** [comment]
This is how upstream names the binary. You'll need to open an issue with them and convince them to start distributing it under a different name.

**Reply by mateusfccp** [reply]
I though maybe Homebrew could have a way of renaming it and dealing with the renamed file?

**Comment by daeho-ro** [comment]
You may open a PR to add the app target with limited cases, https://docs.brew.sh/Cask-Cookbook#stanza-app.
Btw, I don't know what happen after the app is renamed.
When the previously installed users tried to update the application, then it could be failed.

**Reply by mateusfccp** [reply]
Maybe it would fall in this "aesthetic reasons"?

<img width="650" alt="image" src="https://github.com/Homebrew/discussions/assets/4605213/f0cd7ff5-672b-4f50-83e4-f7e3dc7adf92">

**Reply by zlicdt** [reply]
I tried to look through a lot commit history of cask and the TeXstudio repo but still can't find a reason except aesthetic.
Though I really want to change it...
Before version 4.8.5, there have no issue on intel machine, but now the naming is unified on two platforms - `texstudio-version-osx-arch`
